frontend/packages/dev-console/src/components/pipelineruns/logs/log-snippet-types.ts

```diff
@@ -0,0 +1,24 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { Condition } from '../../../utils/pipeline-augment';
+
+type ErrorDetails = {
+  title: string;
+};
+
+export type ErrorDetailsWithLogName = ErrorDetails & {
+  containerName: string;
+  podName: string;
+};
+export type ErrorDetailsWithStaticLog = ErrorDetails & {
+  staticMessage: string;
+};
+
+export type CombinedErrorDetails = ErrorDetailsWithLogName | ErrorDetailsWithStaticLog;
+
+export type LogSnippetTask = K8sResourceKind & {
```

Comment:
Nit... My only reasoning for this would be so that it's metadata the reader of the code can gleam and know it's not all that K8s has to offer. 
```suggestion
type LoggablePipelineKinds = PipelineRun | TaskRunKind;
export type LogSnippetTask = LoggablePipelineKinds & {
```

Reply:
Pretty sure this is moot now based on one of my comments, but I want to leave it for clarity and if I am wrong, I still want to report the Nit 😄

Reply:
It won't work as the earlier implementation of TaskRunKind is all messy